File: rails_app/app/controllers/bench_controller.rb
class BenchController < ApplicationController
  # No sessions or CSRF for these routes, like the other apps.
  skip_forgery_protection

  def plaintext
    render plain: "Hello, World!"
  end

  def json
    render json: { message: "Hello, World!" }
  end

  def fortunes
    rows = Fortune.pluck(:id, :message).map { |id, message| [id, message] }
    rows << [0, "Additional fortune added at request time."]
    @fortunes = rows.sort_by { |_, message| message }
  end
end

File: app/bench_controller.py
from proper import current
from proper.controller import Controller
from .models import Fortune


router = current.app.router


class BenchController(Controller):
    @router.get("plaintext")
    def plaintext(self):
        return self.render(text="Hello, World!")

    @router.get("json")
    def json(self):
        return self.render(json={"message": "Hello, World!"})

    @router.get("fortunes")
    def fortunes(self):
        rows = list(Fortune.select().dicts())
        rows.append({"id": 0, "message": "Additional fortune added at request time."})
        rows.sort(key=lambda r: r["message"])
        self.fortunes = rows
